cpp/oneapi/dal/table/backend/interop/host_homogen_table_adapter.hpp

```diff
@@ -0,0 +1,325 @@
+        status_to_exception(internal_stat);
+        return result;
+    }
+
+private:
+    status_t getBlockOfRows(std::size_t vector_idx,
+                            std::size_t vector_num,
+                            rw_mode_t rwflag,
+                            block_desc_t<double>& block) override {
+        return read_rows_impl(vector_idx, vector_num, rwflag, block);
+    }
+    status_t getBlockOfRows(std::size_t vector_idx,
+                            std::size_t vector_num,
+                            rw_mode_t rwflag,
+                            block_desc_t<float>& block) override {
+        return read_rows_impl(vector_idx, vector_num, rwflag, block);
+    }
+    status_t getBlockOfRows(std::size_t vector_idx,
+                            std::size_t vector_num,
+                            rw_mode_t rwflag,
+                            block_desc_t<int>& block) override {
+        return read_rows_impl(vector_idx, vector_num, rwflag, block);
+    }
+
+    status_t getBlockOfColumnValues(std::size_t feature_idx,
+                                    std::size_t vector_idx,
+                                    std::size_t value_num,
+                                    rw_mode_t rwflag,
+                                    block_desc_t<double>& block) override {
+        return read_column_values_impl(feature_idx, vector_idx, value_num, rwflag, block);
+    }
+    status_t getBlockOfColumnValues(std::size_t feature_idx,
+                                    std::size_t vector_idx,
+                                    std::size_t value_num,
+                                    rw_mode_t rwflag,
+                                    block_desc_t<float>& block) override {
+        return read_column_values_impl(feature_idx, vector_idx, value_num, rwflag, block);
+    }
+    status_t getBlockOfColumnValues(std::size_t feature_idx,
+                                    std::size_t vector_idx,
+                                    std::size_t value_num,
+                                    rw_mode_t rwflag,
+                                    block_desc_t<int>& block) override {
+        return read_column_values_impl(feature_idx, vector_idx, value_num, rwflag, block);
+    }
+
+    status_t releaseBlockOfRows(block_desc_t<double>& block) override {
+        block.reset();
+        return status_t();
+    }
+    status_t releaseBlockOfRows(block_desc_t<float>& block) override {
+        block.reset();
+        return status_t();
+    }
+    status_t releaseBlockOfRows(block_desc_t<int>& block) override {
+        block.reset();
+        return status_t();
+    }
+
+    status_t releaseBlockOfColumnValues(block_desc_t<double>& block) override {
+        block.reset();
+        return status_t();
+    }
+    status_t releaseBlockOfColumnValues(block_desc_t<float>& block) override {
+        block.reset();
+        return status_t();
+    }
+    status_t releaseBlockOfColumnValues(block_desc_t<int>& block) override {
```

Comment:
Isn't it easier to introduce smth like:
`
status_t reset_impl() {
    block.reset();
    return status_t();
}
`
And then use it?

Reply:
The copy-paste just of two lines 😄 
If you insist, I can extract such a method, but do not see the reason in this particular case.